subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/DefaultTransformationRegistrationFactory.java

```diff
@@ -90,36 +91,40 @@ public DefaultTransformationRegistrationFactory(
     public ArtifactTransformRegistration create(ImmutableAttributes from, ImmutableAttributes to, Class<? extends TransformAction> implementation, @Nullable Object parameterObject) {
         List<String> validationMessages = new ArrayList<>();
         TypeMetadata actionMetadata = actionMetadataStore.getTypeMetadata(implementation);
-        actionMetadata.collectValidationFailures(null, new DefaultParameterValidationContext(validationMessages));
-        if (!validationMessages.isEmpty()) {
-            throw new DefaultMultiCauseException(
-                String.format(validationMessages.size() == 1 ? "A problem was found with the configuration of %s." : "Some problems were found with the configuration of %s.", ModelType.of(implementation).getDisplayName()),
-                validationMessages.stream().map(InvalidUserDataException::new).collect(Collectors.toList()));
-        }
+        DefaultParameterValidationContext parameterValidationContext = new DefaultParameterValidationContext(validationMessages);
+        actionMetadata.collectValidationFailures(null, parameterValidationContext);
+        boolean cacheable = implementation.isAnnotationPresent(CacheableTransformAction.class);
 
         // Should retain this on the metadata rather than calculate on each invocation
-        Class<? extends FileNormalizer> inputArtifactNormalizer = AbsolutePathInputNormalizer.class;
-        Class<? extends FileNormalizer> dependenciesNormalizer = AbsolutePathInputNormalizer.class;
+        Class<? extends FileNormalizer> inputArtifactNormalizer = null;
+        Class<? extends FileNormalizer> dependenciesNormalizer = null;
```

Comment:
Why is using `null` here and then replacing it with default better than starting with the default already?

Reply:
The captured `normalizer` can be `null`, so we need to set the default afterwards anyway.